quelpa.el

```diff
@@ -1606,13 +1624,27 @@ Return t in each case."
 
 (defun quelpa-update-cache (cache-item)
   ;; try removing existing recipes by name
-  (setq quelpa-cache (cl-remove (car cache-item)
-                                quelpa-cache :key #'car))
+  (setq quelpa-cache (cl-remove-if
+                      (lambda (item) (eq (car cache-item) (car item)))
+                      quelpa-cache))
   (push cache-item quelpa-cache)
   (setq quelpa-cache
         (cl-sort quelpa-cache #'string<
                  :key (lambda (item) (symbol-name (car item))))))
 
+(defun quelpa-remove-cache (cache-item)
+  "Remove cache-item from `quelpa-cache'."
+  (let ((cache-version (plist-get (cdr cache-item) :version)))
+    (setq quelpa-cache
+          (cl-remove-if
+           (lambda (item)
+             (and (eq (car cache-item) (car item))
+                  (or (not cache-version)
+                      (not (plist-get (cdr item) :version))
+                      (eq cache-version
```

Comment:
Forgive my ignorance: is `cache-version` a string or...?

Reply:
`cache-version` is a list return from package description.

Reply:
Since it's comparing a list, using `equal` is probably a better choice.

Reply:
Done.